leshan-core-cf/src/main/java/org/eclipse/leshan/core/californium/DefaultEndpointFactory.java

```diff
@@ -39,6 +41,11 @@
  */
 public class DefaultEndpointFactory implements EndpointFactory {
 
+    static {
+        CoapConfig.register();
+        DtlsConfig.register();
+    }
```

Comment:
If you ensure, that all applications do it, yes. Otherwise it must be ensured, that this is done before the Californum3.properties are read. That is unfortunately one of the downsides also in the MapBasedEndpointContext.Attributes: the Definition must be in place when reading the values.

Reply:
I search but I didn't find where the definition was added in `MapBasedEndpointContext.Attributes` ?

Reply:
[DtlsEndpointContext.KEY_SESSION_ID](https://github.com/eclipse/californium/blob/master/element-connector/src/main/java/org/eclipse/californium/elements/DtlsEndpointContext.java#L37) passes the Definitions group (`MapBasedEndpointContext.ATTRIBUTE_DEFINITIONS`).
The implies, that the class was loaded implicit before loading attributes.

Reply:
I get it now. Without surprise, I'm not comfortable with this kind of implementation too. For now I haven't idea to make it better.